lib/straight-server/gateway.rb

```diff
@@ -32,15 +32,21 @@ def initialize_blockchain_adapters
       @blockchain_adapters = []
       StraightServer::Config.blockchain_adapters.each do |a|
 
-        adapter = Straight::Blockchain.const_get("#{a}Adapter")
+        adapter = Straight::Blockchain.const_get("#{a['name']}Adapter")
         next unless adapter
-        begin
-          main_url = StraightServer::Config.__send__("#{a.downcase}_url") rescue next
-          test_url = StraightServer::Config.__send__("#{a.downcase}_test_url") rescue nil
-          @blockchain_adapters << adapter.mainnet_adapter(main_url: main_url, test_url: test_url)
-        rescue ArgumentError
-          @blockchain_adapters << adapter.mainnet_adapter
-        end
+
+        @blockchain_adapters <<
+          case a['name']
+          when 'ChainCom'
+            adapter.mainnet_adapter(api_key_id: a['api_key_id'])
+          when 'Insight'
+            main_url = StraightServer::Config.__send__("#{a['name'].downcase}_url")
+            test_url = StraightServer::Config.__send__("#{a['name'].downcase}_test_url")
+            adapter.mainnet_adapter(main_url: main_url, test_url: test_url)
+          else
+            adapter.mainnet_adapter
+          end
```

Comment:
That's getting ugly. Probably, we should use nested hash for adapters configuration, and just pass it with `**config[adapter.name]` without any `case`s. Also, `mainnet_adapter` doesn't look like a right name, because it acts more like some `configure` and even creates shared state — https://github.com/MyceliumGear/straight/blob/master/lib/straight/blockchain_adapters/insight_adapter.rb#L9